Gems/Multiplayer/Code/Source/Components/LocalPredictionPlayerInputComponent.cpp

```diff
@@ -210,15 +216,16 @@ namespace Multiplayer
         m_lastInputReceived = inputArray;
         SetLastInputId(m_lastInputReceived[0].GetClientInputId()); // Set this variable in case of migration
 
-        while (m_lastClientInputId < clientInputId)
+        // Since id values can wrap around, we intentionally compare with a "!=" instead of a "<".
+        while (m_lastClientInputId != clientInputId)
```

Comment:
This always makes me a little nervous, since if we have a double-increment during iteration for some reason this could loop forever.. if we wanted to be more defensive then we could possibly use a sequence check here as well to control loop termination.

Reply:
I'd personally be _more_ worried about letting lastClientInputId grow larger than clientInputId with defensive code than about a bug causing an infinite loop. The infinite loop will produce an obvious symptom that needs to be fixed. Letting the lastClientInputId get too large will cause some number of inputs to get invisibly dropped and/or desync corrected.